src/shape.cpp

```diff
@@ -0,0 +1,295 @@
+        if (dim.isStatic()) {
+            shape.push_back(ov::Dimension(dim.getStaticValue()));
+        } else {
+            shape.push_back(ov::Dimension{dim.getMinValue(), dim.getMaxValue()});
+        }
+    }
+
+    return shape;
+}
+
+shape_t Shape::getFlatShape() const {
+    shape_t shape;
+    shape.reserve(this->dimensions.size());
+
+    for (const Dimension& dim : this->dimensions) {
+        if (dim.getAnyValue() == DYNAMIC_DIMENSION)
+            shape.emplace_back(0);
+        else
+            shape.emplace_back(dim.getAnyValue());
+    }
+
+    return shape;
+}
+
+bool Shape::operator==(const Shape& rhs) const {
+    if (this->dimensions.size() != rhs.dimensions.size())
+        return false;
+
+    for (size_t i = 0; i < this->dimensions.size(); i++) {
+        if (this->dimensions[i] != rhs.dimensions[i]) {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool Shape::operator!=(const Shape& rhs) const {
+    return !(this->operator==(rhs));
+}
+
+std::string Shape::toString() const {
+    std::stringstream ss;
+
+    ss << "(";
+
+    for (size_t i = 0; i < this->dimensions.size(); i++) {
+        ss << this->dimensions[i].toString();
+        if (i < this->dimensions.size() - 1) {
+            ss << ",";
+        }
+    }
+
+    ss << ")";
+
+    return ss.str();
+}
+
+Status Shape::fromString(const std::string& strIn, Shape& shapeOut) {
+    Shape shape;
+    std::string str = strIn;
+
+    erase_spaces(str);
+    if (str.find_first_not_of("0123456789(),-:") != std::string::npos)
+        return StatusCode::SHAPE_WRONG_FORMAT;
+
+    if (std::count(str.begin(), str.end(), '(') != 1)
+        return StatusCode::SHAPE_WRONG_FORMAT;
+
+    if (std::count(str.begin(), str.end(), ')') != 1)
+        return StatusCode::SHAPE_WRONG_FORMAT;
+
+    if (str.size() <= 2)
+        return StatusCode::SHAPE_WRONG_FORMAT;
+
+    if (str.front() != '(' || str.back() != ')')
+        return StatusCode::SHAPE_WRONG_FORMAT;
+
+    str.pop_back();
+    str.erase(str.begin());
+
+    std::vector<std::string> tokens = tokenize(str, ',');
+
+    for (std::string token : tokens) {
+        size_t count = std::count(token.begin(), token.end(), '-');
+        if (count > 1) {
+            SPDLOG_ERROR("Parsing model shape string: {}; too many '-' characters", token);
+            return StatusCode::SHAPE_WRONG_FORMAT;
+        } else if (count == 1 && !token.empty() && *token.begin() != '-') {
+            SPDLOG_ERROR("Parsing model shape string: {}; invalid '-' position", token);
+            return StatusCode::SHAPE_WRONG_FORMAT;
+        }
+
+        count = std::count(token.begin(), token.end(), ':');
+        if (count > 1) {
+            SPDLOG_ERROR("Parsing model shape string: {}; too many ':' characters", token);
+            return StatusCode::SHAPE_WRONG_FORMAT;
+        }
+        try {
+            if (count == 0) {
+                int dimValue = std::stoi(token);
```

Comment:
Lack of exception handling/pre-checking

Reply:
We handle exceptions below: `onst std::out_of_range& e`

Reply:
and `catch (...)`